contracts/chequebook/contract/chequebook.sol

```diff
@@ -21,26 +23,23 @@ contract chequebook is mortal {
         uint8 sig_v, bytes32 sig_r, bytes32 sig_s) {
         // Check if the cheque is old.
         // Only cheques that are more recent than the last cashed one are considered.
-        if(amount <= sent[beneficiary]) return;
+        require(amount > sent[beneficiary]);
         // Check the digital signature of the cheque.
-        bytes32 hash = sha3(address(this), beneficiary, amount);
-        if(owner != ecrecover(hash, sig_v, sig_r, sig_s)) return;
+        bytes32 hash = keccak256(address(this), beneficiary, amount);
+        require(owner == ecrecover(hash, sig_v, sig_r, sig_s));
```

Comment:
These two `require` changes change behaviour: before if the owner/amount didn't match, it resulted in a successful execution without doing anything; now it results in a failed execution using the `revert` opcode (remainder gas is not consumed).

Reply:
Is that good or bad @nagydani ?

Reply:
I think, it is okay. The actual effect is the same: nothing changes in the state and remaining gas is not consumed.